### PR TITLE
GUI Editor Line Gizmo Improvements

### PR DESCRIPTION
**Pull request by carolhmj**
This PR introduces functionality to move, drag endpoints and rotate the Lines Gizmo on the GUI Editor.

Without the change:
current-line-gizmo-behavior

With the change:
new-line-gizmo-behavior

### CODE REVIEW

File: packages/tools/guiEditor/src/diagram/gizmoLine.tsx [hunk 1]
```diff
@@ -2,22 +2,71 @@ import * as React from "react";
...
 import type { Line } from "gui/2D/controls/line";
+import { CoordinateHelper } from "./coordinateHelper";
+import { Matrix2D, MathTools } from "gui/2d/math2D";
 
 interface IGizmoLineProps {
     globalState: GlobalState;
     control: Line;
 }
 
+function getPivot(x1: number, y1: number, x2: number, y2: number, centerX: number, centerY: number) {
```
Comment (RaananW) [inline packages/tools/guiEditor/src/diagram/gizmoLine.tsx:15]: would it make sense making this a `ToRef` function and reuse the vector?

File: packages/tools/guiEditor/src/diagram/gizmoLine.tsx [hunk 2]
```diff
@@ -2,22 +2,71 @@ import * as React from "react";
 import type { GlobalState } from "../globalState";
 import type { IScalePoint } from "./gizmoScalePoint";
 import { GizmoScalePoint, ScalePointPosition } from "./gizmoScalePoint";
-import { CoordinateHelper } from "./coordinateHelper";
 import { Vector2 } from "core/Maths/math";
 import type { Line } from "gui/2D/controls/line";
+import { CoordinateHelper } from "./coordinateHelper";
+import { Matrix2D, MathTools } from "gui/2d/math2D";
```
Comment (RaananW) [inline packages/tools/guiEditor/src/diagram/gizmoLine.tsx:8]: should be `2D` (linux complains, windows is fine ;-) )

File: packages/tools/guiEditor/src/diagram/gizmoLine.tsx [hunk 3]
```diff
@@ -32,45 +81,224 @@ export function GizmoLine(props: IGizmoLineProps) {
...
+            control.y1 = MathTools.Round(y1);
+        } else if (movedPointIndex === 2) {
+            let x2 = control._x2.getValue(control._host);
+            x2 += deltaX;
+            control.x2 = MathTools.Round(x2);
+
+            let y2 = control._y2.getValue(control._host);
+            y2 += deltaY;
+            control.y2 = MathTools.Round(y2);
+        }
```
Comment (RaananW) [inline packages/tools/guiEditor/src/diagram/gizmoLine.tsx:192]: Do you think this code can somehow be reused? or is it too scattered?

File: packages/tools/guiEditor/src/diagram/gizmoLine.tsx [hunk 4]
```diff
@@ -32,45 +81,224 @@ export function GizmoLine(props: IGizmoLineProps) {
...
+    React.useEffect(() => {
+        const pointerMoveObserver = props.globalState.onPointerMoveObservable.add(onMove);
+        const pointerUpObserver = props.globalState.onPointerUpObservable.add(onUp);
+        return () => {
+            props.globalState.onPointerMoveObservable.remove(pointerMoveObserver);
+            props.globalState.onPointerUpObservable.remove(pointerUpObserver);
+        };
+    }, []);
+
+    const onDrag = (event?: React.PointerEvent<HTMLDivElement>, scalePoint?: IScalePoint) => {
```
Comment (RaananW) [inline packages/tools/guiEditor/src/diagram/gizmoLine.tsx:241]: This function is probably being called very often, but it created many objects on each call. I know this is a final product and not the core framework itself, but we should stick to our best practices and reuse as much as possible, where possible.

**Review by RaananW — COMMENTED** [review]
Love it! great work. Just a few comments and a build issue :-)

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by carolhmj** [comment]
@RaananW thanks for the comments, updated the code based on them! :D

**Review by RaananW — APPROVED** [review]

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13394/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13394/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13394/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13394/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13394/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13394/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13394/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13394/merge#BCU1XR#0

**Pull request merged**